added ammo storages to ships and did the lay-outs a lot (#2269 retry)

KiLEdEnNis opened the pull request:
I retried manually copy-pasting the 5 files in my fork instead and made it a much cleaner PR than #2269 (same PR though), closing that one and this will be my PR

copy-pasted over what I  did:
- added ammo storages to some ships that had missiles and room left (only the stock Protector lost 2 of the 3 small radar jammers for 2 torpedo storages), also added the ammo for them
- made a lot of lay-out fixes and improvements
- ordened all variants in variants.txt alphabetically & even ordened the variants per ship), and put all Hai outfitted human ship variants to the bottom seperately
- had to manually add the hardpoint fix commit but that's 100% fine now

as far as I know I made 0 mistakes (perfectionism) but please tell me if I made any (found 1 myself: the base Bastion didn't have extra ammo for the boxes, re-checked everything and didn't see any mistakes myself)
edit: also I manually calculated the remaining outfit space on every ship and variant I changed, so I'm 99% sure none have negative outfit space

Amazinite commented on data/korath ships.txt:
```diff
@@ -1118,7 +1131,8 @@ ship "Model 256"
 		"Korath Minelayer" 2
-		"Korath Mine" 34
+		"Korath Mine" 52
+		"Korath Mine Rack" 2
```
Not *more* mines!
What have I created?

KiLEdEnNis replied:
well, storages got added so had to add some of them :P

Amazinite commented on data/variants.txt:
```diff
@@ -8,280 +8,225 @@
-		"Chipmunk Plasma Steering"
-		"Hyperdrive"
-
+# Human variants (other list is Alien variants; Hai outfitted ships).
```
Is the comment really necessary?

KiLEdEnNis replied:
I can delete it if it matters, it's just to make it clear

jafdy commented:
I have no idea what you did with variants.txt but I can't follow it. I think that a reformatting PR should not be in the same place as a PR with actual changes.

KiLEdEnNis commented on data/ships.txt:
```diff
@@ -334,6 +335,7 @@ ship "Bastion"
 		"Plasma Cannon" 2
 		"Heavy Rocket Launcher" 2
 		"Heavy Rocket" 40
```
found my 1st mistake

jafdy replied:
Well fix it then.

KiLEdEnNis replied:
did so

Amazinite commented on data/variants.txt:
```diff
@@ -1473,108 +1668,62 @@ ship "Mule" "Mule (Hai Engines)"
-		"Hyperdrive"
-
-
+# The Osprey (Alien Weapons) isn't used in any fleets, only in a special bounty mission.
```
I don't really see the point in the comments.

KiLEdEnNis replied:
because I put that Osprey & the Behemoth (Heavy) in the lists while they were at like the bottom; these 2 don't appear in real fleets so was to make it clear
I can delete this & the Behemoth (Heavy) lines, those 2 + the human & alien indicator ones are the only actual comment lines I typed

Amazinite commented:
👌

KiLEdEnNis commented on data/ships.txt:
```diff
@@ -947,7 +951,8 @@ ship "Cruiser"
 		"Sidewinder Missile Launcher" 2
-		"Sidewinder Missile" 100
+		"Sidewinder Missile" 250
+		"Sidewinder Storage Rack" 6
```
about the biggest ammo buff I made
if someone thinks it's a really big deal I can nerf it to 4, but it's a warship after all

KiLEdEnNis commented on data/ships.txt:
```diff
@@ -2264,7 +2274,8 @@ ship "Rainmaker"
 		"Meteor Missile Launcher" 4
-		"Meteor Missile" 140
+		"Meteor Missile" 284
+		"Meteor Missile Box" 8
```
next to the Cruiser the biggest storage buff; double the missiles, double the fun
I can bring it back to 6 or 4 but I prefer to not go below 4

KiLEdEnNis commented on data/variants.txt:
```diff
@@ -1144,134 +1150,359 @@ ship "Roost" "Roost (Sidewinder)"
+
+
+
+# Alien variants.
```
same here @Amazinite I can delete it if it matters, it's just to make it clear

KiLEdEnNis commented:
@jafdy I kinda already did, so if I made a separate reformatting PR they'd include the ammo changes as well

all I did is order them alphabetically, and ships w/ multiple variants have the variant names ordered alphabetically as well

KiLEdEnNis commented on data/wanderer ships.txt:
```diff
@@ -562,7 +566,8 @@ ship "Deep River"
 		"Thunderhead Launcher" 6
-		"Thunderhead Missile" 240
+		"Thunderhead Missile" 320
+		"Thunderhead Storage Array" 4
```
thinking about removing one for the sake of OCD: the last volley has 2 missiles that shoot when it has 6 launchers
it would also pair with balancing a bit

jafdy commented:
It's good practice to make a new branch for each PR insted of bringing it from the master.

KiLEdEnNis commented:
yes but I'd need to undo & redo everything to make 2 separate PR's
as for the 4 # comments, gonna remove them as others don't think they're needed

KiLEdEnNis commented:
so that's all; removed the #'s, fixed a mistake & removed a thunderhead launcher from the Deep River so the last volley uses all launchers
all I need now is people reviewing it if I have to remove ammo boxes somewhere

MessyMix commented:
tfw you desperately buff your contributing count
+1.2k lines, -1k lines

xDD

KiLEdEnNis commented:
"foxed" end my life
the biggest ship change I made; added 2 expansions, next to the Protector losing 2 small radar jammers, the rest is ammo storages & lay-out change

KiLEdEnNis commented on data/hai ships.txt:
```diff
@@ -300,12 +315,31 @@ ship "Shield Beetle" "Shield Beetle (Pulse)"
+		"Boulder Reactor"
+		"Geode Reactor"
+		"Hai Valley Batteries"
+		"Mass Expansion" 2
```
added the 2 expansions to fit 4 storages, I can take 1 out to half the storages it has?

KiLEdEnNis commented:
well that was a waste of 10+ hours
have fun to the next one who may add ammo storages

KiLEdEnNis commented:
once the other ship-file-changing PR's get merged/denied, would it be worth to pick this PR up again (in a new PR)?
I can still see my changes here so I could add ammo boxes to missile ships, and if that gets merged I could revisit the "reformatting" part as well (improve the lay-out in all ship files)
